cou/steps/__init__.py

```diff
@@ -221,3 +230,61 @@ async def run(self) -> Any:
             return await self._task  # wait until task is completed
         except asyncio.CancelledError:  # ignoring asyncio.CancelledError
             logger.warning("Task %s was stopped unsafely.", repr(self))
+
+
+class ApplicationUpgradeStep(BaseStep):
+    """Represents the step for application-level upgrade."""
+
+
+class UpgradeSubStep(BaseStep):
+    """Represents the sub-step of an application upgrade."""
+
+
+class PreUpgradeSubStep(UpgradeSubStep):
+    """Represents the pre-upgrade sub-step for an application upgrade."""
+
+    @property
+    def description(self) -> str:
+        """Get the description of the PreUpgradeSubStep.
+
+        :return: description
+        :rtype: str
+        """
+        return self._description
+
+    @description.setter
+    def description(self, description: str) -> None:
+        """Set the description of the PreUpgradeSubStep.
+
+        :param description: description
+        :type description: str
+        :raises ValueError: When a coroutine is passed without description.
+        """
+        if not description and self._coro:
+            raise ValueError("Every coroutine should have a description")
+        self._description = "[pre-upgrade] " + description
```

Comment:
if you check my draft PR #165, there is better solution to do overwrite the setter from base class or we can go with what @gabrielcocenza suggesting, only moving it to base class or defining `UpgradeStep` instead of `UpgradeSubStep`.